### PR TITLE
refactor: replace namespace with workspace-namespace metadata

### PR DESCRIPTION
**Pull request by amandavialva01**
## Ticket
DET-10377

## Description
Requests should take in all metadata about a workspace-namespace binding (namespace name, whether we want to auto-create a namespace, optional resource quota placed on the namespace) packed into a singular struct, and should return workspace-namespace metadata as such.

## Test Plan
Refactor, so just make sure CI passes.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by carolinaecalderon — APPROVED** [review]
master changes LGTM; deferring to @jgongd for the harness & proto changes

also, make sure to lint before merging!

File: proto/src/determined/workspace/v1/workspace.proto [hunk 1]
```diff
@@ -97,18 +95,42 @@ message PatchWorkspace {
...
+  // bound to the workspace for a given cluster).
+  map<string, determined.workspace.v1.WorkspaceNamespaceMeta>
+      cluster_namespace_meta = 18;
 }
 
 // WorkspaceNamespace represents a workspace-namespace binding for a given
-// cluster.
-message WorkspaceNamespace {
+// workspace and cluster.
+message WorkspaceNamespaceBinding {
```
Comment (salonig23) [inline proto/src/determined/workspace/v1/workspace.proto:107]: can we add auto_namespace to this and return this in ListWorkspaceNamespaceBindingsEndPoint? Alternatively, return WorkspaceNamespaceMeta

File: webui/react/src/types.ts [hunk 2]
```diff
@@ -1059,7 +1059,7 @@ export interface Workspace {
 }
 
 export interface WorkspaceNamespaceBindings {
-  clusterNamespacePairs: Record<string, string>;
+  clusterNamespacePairs: { [key: string]: Api.V1WorkspaceNamespaceBinding; };
```
Comment (keita-determined) [inline webui/react/src/types.ts:1062]: can we use `Record`?

**Review by keita-determined — APPROVED** [review]

File: webui/react/src/types.ts (hunk 2 above)
Comment (amandavialva01) [reply webui/react/src/types.ts]: Done!

**Review by jgongd — APPROVED** [review]
LGTM!

**Pull request merged**